packages/babel-parser/src/plugins/flow/index.js

```diff
@@ -1910,20 +1910,22 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       expr: N.Expression,
       startPos: number,
       startLoc: Position,
-      refNeedsArrowPos?: ?Pos,
+      refExpressionErrors?: ?ExpressionErrors,
     ): N.Expression {
       if (!this.match(tt.question)) return expr;
 
       // only use the expensive "tryParse" method if there is a question mark
       // and if we come from inside parens
-      if (refNeedsArrowPos) {
+      if (this.state.maybeInArrowParameters) {
         const result = this.tryParse(() =>
           super.parseConditional(expr, startPos, startLoc),
         );
 
         if (!result.node) {
-          // $FlowIgnore
-          refNeedsArrowPos.start = result.error.pos || this.state.start;
+          if (refExpressionErrors) {
```

Comment:
If `this.state.maybeInArrowParameters` is `true`, `refExpressionErrors` must be non-nullish. We can use instruct flow that `refExpressionErrors` is non-null:

```
/*:: invariant(refExpressionErrors != null) */
```

Ditto for `result.error`, if `result.code` is `null`, `result.error` is always non-nullish.